tests/cpp/lp_namer/MPSFileProblemProviderAdapterTest.cpp

```diff
@@ -0,0 +1,53 @@
+#include <fstream>
+#include <RandomDirGenerator.h>
+
+
+#include "gtest/gtest.h"
+
+#include "antares-xpansion/lpnamer/problem_modifier/MPSFileProblemProviderAdapter.h"
+
+class FixtureMPSFileProblemProviderAdapter : public ::testing::Test
+{
+    void SetUp() override
+    {
+        previous_path = std::filesystem::current_path();
+        tmp_dir = CreateRandomSubDir(std::filesystem::temp_directory_path());
+        std::filesystem::current_path(tmp_dir);
+    }
+
+    void TearDown() override
+    {
+        std::filesystem::current_path(previous_path);
+    }
+
+    std::filesystem::path previous_path;
+public:
+    std::filesystem::path tmp_dir;
+};
+
+TEST_F(FixtureMPSFileProblemProviderAdapter, ProvideProblemProperly) {
+        std::ofstream file(tmp_dir / "problem_1_1.mps");
```

Comment:
Delete this file afterwards ?

Reply:
It's in temp. It's best to keep them for debugging purpose.